src/azul/chalice.py

```diff
@@ -74,16 +74,19 @@ def __init__(self,
         else:
             self._specs: Optional[MutableJSON] = None
         super().__init__(app_name, debug=config.debug > 0, configure_logs=False)
-        self.register_middleware(self.logging_middleware, 'http')
-        self.register_middleware(self.authentication_middleware, 'http')
+        self.register_middleware(self._logging_middleware, 'http')
+        # SKETCH: Not sure about the ordering
```

Comment:
I think this should be the first middleware that's being *invoked*. How to configure that, I don't know.

Reply:
Experiments and looking at the source code indicate that the order of invocation matches the order of registration.